### PR TITLE
refactor!: to esm, common eslint config, @exodus/bitcoinjs and @exodus/crypto

### PR DESCRIPTION
**Pull request by mvayngrib**
depends on #3 

review commit by commit. The big diff is from lint autofix and deps

probably a breaking change

## Test Plan

- [x] tests pass
- [x] `node payreq.js` runs without errors (ESM)

### CODE REVIEW

File: payreq.js [hunk 1]
```diff
@@ -193,40 +188,43 @@ function convert (data, inBits, outBits) {
   return result
 }
 
-function wordsToBuffer (words, trim) {
-  let buffer = Buffer.from(convert(words, 5, 8, true))
```
Comment (mvayngrib) [inline payreq.js:197]: fix: convert only accepts 3 args

File: payreq.js [hunk 2]
```diff
@@ -468,52 +492,59 @@ function millisatToHrp (millisatoshis) {
...
   }
 
   return outputString ? millisatoshisBN.toString() : millisatoshisBN
 }
 
-function sign (inputPayReqObj, inputPrivateKey) {
-  const payReqObj = cloneDeep(inputPayReqObj)
-  const privateKey = hexToBuffer(inputPrivateKey)
+export function sign(inputPayReqObj, inputPrivateKey) {
+  const payReqObj = Object.assign(Object.create(null), cloneDeep(inputPayReqObj))
```
Comment (mvayngrib) [inline payreq.js:539]: fix: this was incorrectly returning an Object prototype

File: package.json [hunk 3]
```diff
@@ -11,16 +12,18 @@
...
   },
   "devDependencies": {
-    "nyc": "^15.0.0",
-    "standard": "*",
-    "tap-dot": "*",
-    "tape": "^4.13.2"
+    "@exodus/eslint-config-exodus": "^1.0.1",
+    "@exodus/prettier": "^1.0.0",
+    "@exodus/test": "^1.0.0-rc.57",
+    "eslint": "^8.57.1",
```
Comment (fboucquez) [inline package.json]: nit: eslint v9 like hydra

**Review by fboucquez — APPROVED** [review]
utACK

Thanks! 

Question, What is the diverge threshold where this lib is not a fork anymore, and it should be moved to assets or hydra? Not in a rush to do it, but just to consider it

File: package.json (hunk 3 above)
Comment (mvayngrib) [reply package.json]: bumped 2721368e29e1e8b38d6103becec7dadc5f2b4812

**Comment by mvayngrib** [comment]
> What is the diverge threshold were this lib is not a fork anymore and it should be moved to assets or hydra? Not in a rush to do it, but just to consider it

we're definitely past that threshold. I can see them accepting a PR for esm support, butI doubt they'll be interested in using our libs. No rush to move to hydra though

**Comment by fboucquez** [comment]
Good to start merging and publishing? I don't have the auth token with me now to publish, but I can do later today

**Comment by mvayngrib** [comment]
sgtm, we can give @ChALkeR a couple hours in case he wants to take a look

**Comment by fboucquez** [comment]
I'll migrate the repo to assets

**Pull request merged**